File: SDKs/amazonaws.com/cloudsearch/2013-01-01/go/pkg/models/operations/postindexdocuments.go
package operations




type PostIndexDocumentsActionEnum string

const (
    PostIndexDocumentsActionEnumIndexDocuments PostIndexDocumentsActionEnum = "IndexDocuments"
)



type PostIndexDocumentsVersionEnum string

const (
    PostIndexDocumentsVersionEnumTwoThousandAndThirteen0101 PostIndexDocumentsVersionEnum = "2013-01-01"
)


type PostIndexDocumentsQueryParams struct {
    Action PostIndexDocumentsActionEnum `queryParam:"style=form,explode=true,name=Action"`
    Version PostIndexDocumentsVersionEnum `queryParam:"style=form,explode=true,name=Version"`
    
}

type PostIndexDocumentsHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    
}

type PostIndexDocumentsRequest struct {
    QueryParams PostIndexDocumentsQueryParams 
    Headers PostIndexDocumentsHeaders 
    Request []byte `request:"mediaType=text/xml"`
    
}

type PostIndexDocumentsResponse struct {
    Body []byte 
    ContentType string 
    StatusCode int64 
    
}

File: SDKs/amazonaws.com/cloudsearch/2013-01-01/typescript/src/sdk/models/operations/postindexdocuments.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PostIndexDocumentsActionEnum {
    IndexDocuments = "IndexDocuments"
}

export enum PostIndexDocumentsVersionEnum {
    TwoThousandAndThirteen0101 = "2013-01-01"
}


export class PostIndexDocumentsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Action" })
  action: PostIndexDocumentsActionEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Version" })
  version: PostIndexDocumentsVersionEnum;
}


export class PostIndexDocumentsHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;
}


export class PostIndexDocumentsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: PostIndexDocumentsQueryParams;

  @SpeakeasyMetadata()
  headers: PostIndexDocumentsHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=text/xml" })
  request?: Uint8Array;
}


export class PostIndexDocumentsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
